### ISSUE TITLE
Add support for BackendConfig

### ISSUE THREAD

**Issue by larsenqec** [terraform-providers/terraform-provider-google#6504]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

We currently use a BackendConfig object to be able to control timeouts on kubernetes services. By creating a named BackendConfig object, services can reference that name in their annotations thereby overriding the (short) default timeouts. Services that handle large file uploads or other long running processes need longer timeouts. Timeouts are my use-case, but there are other features in BackendConfig.

### New or Affected Resource(s)

* google_compute_backend_config

### Potential Terraform Configuration

```tf
# Propose what you think the configuration to take advantage of this feature should look like.
# We may not use it verbatim, but it's helpful in understanding your intent.

resource google_compute_backend_config my_backend_config {
  metadata {
    name = "my_backend_timeout"
  }
  timeout_sec = 600
  iap = ...
  cdn = ...
  securityPolicy = ...
  connectionDraining = ...
  sessionAffinity = ...
  customRequestHeaders = ...
  logging = ...
}

```

### References

* https://cloud.google.com/kubernetes-engine/docs/concepts/backendconfig
* https://cloud.google.com/kubernetes-engine/docs/how-to/configure-backend-service

**Comment by rileykarson** [comment]
Hey @larsenqec!

I'm not sure that this resource is something that should be managed using this provider. While it's true that it's somewhat of a GCP resource, given that it's a GKE-published Kubernetes custom resource that creates a backend service, it isn't a resource that's [directly] created through the _GCP APIs_.

The `google` (and `google-beta`) providers are aware of how to authenticate with GCP's REST APIs, but to support this (and other K8S custom resources) we'd need to also allow configuring Kubernetes authentication / pointing the provider at a K8S cluster. Definitely possible, but undesirable- one authentication method / api service to provider is the general pattern providers follow.

I think that the best way to configure that custom resource today would be the [new `kubernetes-alpha` provider](https://www.hashicorp.com/blog/deploy-any-resource-with-the-new-kubernetes-provider-for-hashicorp-terraform/) using the `kubernetes_manifest` resource, which was created to handle managing custom resources. Otherwise, the `helm` provider is another alternative.

Has it been your experience that there are a handful of GCP-specific custom resources you want to be able to configure, or is this the only one? One (long-term) possibility would be if we explored a provider with first-class resources for GCP CRs to be used alongside the `kubernetes` provider.

**Comment by larsenqec** [comment]
Yes, I suppose it is complicated if it's not exposed through the Google REST API.  I will look at the kubernetes_manifest resource. Thank you.
